File: apps/client/lib/core/api/models/enums.dart
// Enums matching the Prisma schema

enum Gender {
  MALE,
  FEMALE;

  String get displayName {
    switch (this) {
      case Gender.MALE:
        return 'Male';
      case Gender.FEMALE:
        return 'Female';
    }
  }
}

enum ActivityLevel {
  SEDENTARY,
  LIGHTLY_ACTIVE,
  MODERATELY_ACTIVE,
  ACTIVE,
  VERY_ACTIVE;

  String get displayName {
    switch (this) {
      case ActivityLevel.SEDENTARY:
        return 'Sedentary';
      case ActivityLevel.LIGHTLY_ACTIVE:
        return 'Lightly Active';
      case ActivityLevel.MODERATELY_ACTIVE:
        return 'Moderately Active';
      case ActivityLevel.ACTIVE:
        return 'Active';
      case ActivityLevel.VERY_ACTIVE:
        return 'Very Active';
    }
  }

  double get multiplier {
    switch (this) {
      case ActivityLevel.SEDENTARY:
        return 1.2;
      case ActivityLevel.LIGHTLY_ACTIVE:
        return 1.375;
      case ActivityLevel.MODERATELY_ACTIVE:
        return 1.55;
      case ActivityLevel.ACTIVE:
        return 1.725;
      case ActivityLevel.VERY_ACTIVE:
        return 1.9;
    }
  }
}

enum MealType {
  BREAKFAST,
  LUNCH,
  DINNER,
  SNACK;

  String get displayName {
    switch (this) {
      case MealType.BREAKFAST:
        return 'Breakfast';
      case MealType.LUNCH:
        return 'Lunch';
      case MealType.DINNER:
        return 'Dinner';
      case MealType.SNACK:
        return 'Snack';
    }
  }
}

enum EnergyUnit {
  KCAL,
  KJ;

  String get displayName {
    switch (this) {
      case EnergyUnit.KCAL:
        return 'kcal';
      case EnergyUnit.KJ:
        return 'kJ';
    }
  }
}

enum DisplayMode {
  QUALITATIVE,
  EXACT;

  String get displayName {
    switch (this) {
      case DisplayMode.QUALITATIVE:
        return 'Qualitative';
      case DisplayMode.EXACT:
        return 'Exact';
    }
  }
}

enum FoodDataSource {
  FATSECRET,
  MANUAL,
  OPEN_FOOD_FACTS,
  USDA;
}

enum ApprovalStatus {
  PENDING,
  APPROVED,
  REJECTED;

  String get displayName {
    switch (this) {
      case ApprovalStatus.PENDING:
        return 'Pending';
      case ApprovalStatus.APPROVED:
        return 'Approved';
      case ApprovalStatus.REJECTED:
        return 'Rejected';
    }
  }
}

File: apps/web/src/api/types/enums.ts
// Enums matching the Prisma schema

export enum Gender {
  MALE = 'MALE',
  FEMALE = 'FEMALE',
}

export const GenderDisplay: Record<Gender, string> = {
  [Gender.MALE]: 'Male',
  [Gender.FEMALE]: 'Female',
}

export enum ActivityLevel {
  SEDENTARY = 'SEDENTARY',
  LIGHTLY_ACTIVE = 'LIGHTLY_ACTIVE',
  MODERATELY_ACTIVE = 'MODERATELY_ACTIVE',
  ACTIVE = 'ACTIVE',
  VERY_ACTIVE = 'VERY_ACTIVE',
}

export const ActivityLevelDisplay: Record<ActivityLevel, string> = {
  [ActivityLevel.SEDENTARY]: 'Sedentary',
  [ActivityLevel.LIGHTLY_ACTIVE]: 'Lightly Active',
  [ActivityLevel.MODERATELY_ACTIVE]: 'Moderately Active',
  [ActivityLevel.ACTIVE]: 'Active',
  [ActivityLevel.VERY_ACTIVE]: 'Very Active',
}

export const ActivityLevelMultiplier: Record<ActivityLevel, number> = {
  [ActivityLevel.SEDENTARY]: 1.2,
  [ActivityLevel.LIGHTLY_ACTIVE]: 1.375,
  [ActivityLevel.MODERATELY_ACTIVE]: 1.55,
  [ActivityLevel.ACTIVE]: 1.725,
  [ActivityLevel.VERY_ACTIVE]: 1.9,
}

export enum MealType {
  BREAKFAST = 'BREAKFAST',
  LUNCH = 'LUNCH',
  DINNER = 'DINNER',
  SNACK = 'SNACK',
}

export const MealTypeDisplay: Record<MealType, string> = {
  [MealType.BREAKFAST]: 'Breakfast',
  [MealType.LUNCH]: 'Lunch',
  [MealType.DINNER]: 'Dinner',
  [MealType.SNACK]: 'Snack',
}

export enum EnergyUnit {
  KCAL = 'KCAL',
  KJ = 'KJ',
}

export const EnergyUnitDisplay: Record<EnergyUnit, string> = {
  [EnergyUnit.KCAL]: 'kcal',
  [EnergyUnit.KJ]: 'kJ',
}

export enum DisplayMode {
  QUALITATIVE = 'QUALITATIVE',
  EXACT = 'EXACT',
}

export const DisplayModeDisplay: Record<DisplayMode, string> = {
  [DisplayMode.QUALITATIVE]: 'Qualitative',
  [DisplayMode.EXACT]: 'Exact',
}

export enum FoodDataSource {
  FATSECRET = 'FATSECRET',
  MANUAL = 'MANUAL',
  OPEN_FOOD_FACTS = 'OPEN_FOOD_FACTS',
  USDA = 'USDA',
}

export enum ApprovalStatus {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED',
}

export const ApprovalStatusDisplay: Record<ApprovalStatus, string> = {
  [ApprovalStatus.PENDING]: 'Pending',
  [ApprovalStatus.APPROVED]: 'Approved',
  [ApprovalStatus.REJECTED]: 'Rejected',
}
